plugins/samples/redirect_bulk/plugin.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_redirect_bulk]
+package main
+
+import (
+	"strings"
+
+	"github.com/proxy-wasm/proxy-wasm-go-sdk/proxywasm"
+	"github.com/proxy-wasm/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func main() {}
+
+func init() {
+	proxywasm.SetVMContext(&vmContext{})
+}
+
+type vmContext struct {
+	types.DefaultVMContext
+}
+
+type pluginContext struct {
+	types.DefaultPluginContext
+	domainMappings map[string]string
+}
+
+type httpContext struct {
+	types.DefaultHttpContext
+	domainMappings map[string]string
+}
+
+func (*vmContext) NewPluginContext(uint32) types.PluginContext {
+	return &pluginContext{
+		domainMappings: make(map[string]string),
+	}
+}
+
+func (p *pluginContext) NewHttpContext(uint32) types.HttpContext {
+	return &httpContext{
+		domainMappings: p.domainMappings,
+	}
+}
+
+func (p *pluginContext) OnPluginStart(pluginConfigurationSize int) types.OnPluginStartStatus {
+	// Get plugin configuration
+	config, err := proxywasm.GetPluginConfiguration()
+	if err != nil {
+		proxywasm.LogErrorf("Failed to get plugin configuration: %v", err)
+		return types.OnPluginStartStatusFailed
+	}
+
+	// Parse configuration
+	configStr := string(config)
+	mappings := make(map[string]string)
+
+	// Parse each line as "source_domain target_domain"
+	for _, line := range strings.Split(configStr, "\n") {
```

Comment:
This string conversion and the strings.Split will create a large number of strings that have to be garbage collected later.

Can you use the `bytes.Split(config, []byte("\n"))`, `bytes.TrimSpace()`, and `bytes.Fields`, the convert to string only on line 79 when inserting into the map? That will avoid creating any intermediate strings, and only create views into the single `config` byte slice.